lib/HLSL/DxilValidation.cpp

```diff
@@ -3100,6 +3100,15 @@ static void ValidateFunctionBody(Function *F, ValidationContext &ValCtx) {
             }
             dispatchMesh = CI;
           }
+
+          for (Value *op : CI->operands()) {
+            if (isa<UndefValue>(op)) {
+              if (dxilOpcode != DXIL::OpCode::OutputComplete) {
+                ValCtx.EmitInstrError(
```

Comment:
What about dxil op like sample which use undef for unused channel?

Reply:
Right, this is supposed to catch use of undef or zeroinitializer handle values only.  The only case that's allowed is zeroinitializer (not undef) for OutputComplete which is a no-op.  Also, we could eliminate the zeroinitializer OutputComplete when finalizing IR and still say that it is invalid during validation.